### PR TITLE
[Test][Misc] Fix libsecp256k1 init for unit test setup + Allow to specify blocks storage directory

### PR DESCRIPTION
**Pull request by furszy**
Grouped two different, zero risk, modifications here to not have to create two/three small and straightforward PRs:

#### 1) Unit Tests:

* Fixing basic unit test setup issue, was not initializing the libsecp256k1 context. Thus why we have been forced to use the extended unit testing setup for almost every test case (which includes CConnman, scheduler, validation interface, etc.. that are not needed so often).
* Refactor: `BasicTestingSetup` receiving the network in the constructor.

#### 2) Back ported the ability to specify a custom directory for the blocks storage.

> Since the actual block files taking up more and more space, it may be desirable to have them stored in a different location then the data directory (use case: SSD for chainstate, etc., HD for blocks).
This PR adds a -blocksdir option that allows one to keep the blockfiles external from the data directory (instead of creating symlinks).

* #9895.
* #12653.
* #14364.
* #14409.
* #15124.
* #17059.

### CODE REVIEW

File: test/functional/feature_blocksdir.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+
+    def run_test(self):
+        self.stop_node(0)
+        node0path = os.path.join(self.options.tmpdir, "node0")
+        shutil.rmtree(node0path)
+        initialize_datadir(self.options.tmpdir, 0)
+        self.log.info("Starting with non exiting blocksdir ...")
```
Comment (random-zebra) [inline test/functional/feature_blocksdir.py:24]: nit: s/"non exiting"/"non existing"-"nonexistent"

File: test/functional/feature_blocksdir.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        self.assert_start_raises_init_error(0, ["-blocksdir="+self.options.tmpdir+ "/blocksdir"], "Specified blocks director")
+        os.mkdir(self.options.tmpdir+ "/blocksdir")
+        self.log.info("Starting with exiting blocksdir ...")
```
Comment (random-zebra) [inline test/functional/feature_blocksdir.py:27]: nit:
```suggestion
        self.log.info("Starting with existing blocksdir ...")
```

**Review by random-zebra — COMMENTED** [review]
Looking good. Couple minor nits/typos in the comments. Otherwise ACK.

File: test/functional/feature_blocksdir.py (hunk 2 above)
Comment (furszy) [reply test/functional/feature_blocksdir.py]: done

**Comment by furszy** [comment]
updated, typos corrected.

**Comment by Fuzzbawls** [comment]
functionally good. might as well document the new option in the release notes template here so it doesn't need to be done later

**Comment by furszy** [comment]
Updated per feedback, added `-blocksdir` init option to the release-notes.

File: doc/release-notes.md [hunk 3]
```diff
@@ -44,6 +44,9 @@ Notable Changes
 
 (Developers: add your notes here as part of your pull requests whenever possible)
 
+#### Disable PoW mining RPC Commands
```
Comment (Fuzzbawls) [inline doc/release-notes.md:47]: wrong section title?

**Review by Fuzzbawls — APPROVED** [review]
looks like something went awry when adding to the release notes. minor issue though and can simply be cleaned up later.

otherwise, ACK ff767e11337695a11ea3953378190cd14d6fdf70

File: doc/release-notes.md (hunk 3 above)
Comment (furszy) [reply doc/release-notes.md]: yeah, done. Updated.

**Review by Fuzzbawls — APPROVED** [review]
re-ACK b0e0d554f6be1ec2fecbd1282d39a78eea7a9753

**Review by random-zebra — APPROVED** [review]
utACK b0e0d554f6be1ec2fecbd1282d39a78eea7a9753 and merging...

**Pull request merged**